validation/barotropic_gyre/visualize_barotropic_gyre.jl

```diff
@@ -29,8 +29,8 @@ function visualize_barotropic_gyre(filepath)
 
     file = jldopen(filepath)
 
-    Nx = file["grid/Nx"]
-    Ny = file["grid/Ny"]
+    Nx = file["grid/grid/Nx"]
+    Ny = file["grid/grid/Ny"]
```

Comment:
Ah crepes, that is annoying. It might be worth defining a custom `saveproperty!` for `ImmersedBoundaryGrid`. Work for a future PR.

Reply:
@glwagner should we merge this and then fix later.

P.S. later fixes could also include thinking whether having cube, immersed boundary as very separate types is the best approach or whether there is an alternate approach where the simpler forms (cartesian, non-immersed etc..) are specializations of some general type/data structure.

Reply:
I think fine to merge for now.